### PR TITLE
`Paywalls`: add methods for presenting paywalls with an offering identifier (Android)

### PR DESCRIPTION
**Pull request by vegaro**
Added `newInstance` function that accepts an `offeringIdentifier` instead of an `Offering`

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -223,6 +225,7 @@ jobs:
     executor:
       name: android/android-machine
       resource-class: large
+      tag: default
```
Comment (tonidero) [inline .circleci/config.yml:228]: This is from #669 right? Also, seems that this PR might need a rebase since it has some unrelated changes.

File: android/hybridcommon-ui/src/main/java/com/revenuecat/purchases/hybridcommon/ui/PaywallFragment.kt [hunk 2]
```diff
@@ -27,14 +27,31 @@ internal class PaywallFragment : Fragment(), PaywallResultHandler {
...
+            )
+        }
+
+        @JvmStatic
+        fun newInstance(
+            activity: FragmentActivity,
+            requiredEntitlementIdentifier: String? = null,
+            paywallResultListener: PaywallResultListener? = null,
+            shouldDisplayDismissButton: Boolean? = null,
+            offeringIdentifier: String? = null,
```
Comment (tonidero) [inline android/hybridcommon-ui/src/main/java/com/revenuecat/purchases/hybridcommon/ui/PaywallFragment.kt:46]: I wonder if we should deprecate the old one (and remove it?), I don't think we will be using it since it would be much trickier.

File: android/hybridcommon-ui/src/main/java/com/revenuecat/purchases/hybridcommon/ui/PaywallHelpers.kt [hunk 3]
```diff
@@ -4,25 +4,67 @@ import androidx.fragment.app.FragmentActivity
...
+            )
+            .commit()
+    }
+}
+
+data class PresentPaywallOptions(
+    val requiredEntitlementIdentifier: String? = null,
+    val paywallResultListener: PaywallResultListener? = null,
+    val shouldDisplayDismissButton: Boolean? = null,
+    val offering: Offering? = null,
```
Comment (tonidero) [inline android/hybridcommon-ui/src/main/java/com/revenuecat/purchases/hybridcommon/ui/PaywallHelpers.kt:68]: Hmm could we just entirely remove this one in favor of `offeringIdentifier`? If it's for the deprecated method, we can just pass in the id there I think.

**Review by tonidero — COMMENTED** [review]
Left a couple of comments but it's looking good

File: android/hybridcommon-ui/src/main/java/com/revenuecat/purchases/hybridcommon/ui/PaywallHelpers.kt (hunk 3 above)
Comment (NachoSoto) [reply android/hybridcommon-ui/src/main/java/com/revenuecat/purchases/hybridcommon/ui/PaywallHelpers.kt:68]: Wouldn't that mean we have to fetch offerings again potentially? 

Can we define a type like this:
```kotlin
sealed class PaywallSource {
    class Offering(val value: PurchasesOffering): PaywallSource()
    class DefaultOffering()
    class OfferingIdentifier(val value: String): PaywallSource()
}
```

Like I did in iOS, to make the 3 options type safe?

Comment (tonidero) [reply android/hybridcommon-ui/src/main/java/com/revenuecat/purchases/hybridcommon/ui/PaywallHelpers.kt:68]: Right... Yeah makes sense. We have something similar as an internal API in native android: https://github.com/RevenueCat/purchases-android/blob/main/ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/PaywallOptions.kt#L6... But I don't know if we want to expose that. But yeah, in here that could make sense.

Comment (vegaro) [reply android/hybridcommon-ui/src/main/java/com/revenuecat/purchases/hybridcommon/ui/PaywallHelpers.kt:68]: > Wouldn't that mean we have to fetch offerings again potentially?

The hybrids are always going to use the offering identifier since there's no way to send a whole offering from JS, Dart, etc to the native side

File: .circleci/config.yml (hunk 1 above)
Comment (vegaro) [reply .circleci/config.yml]: sorry, I had the wrong base branch, these are gone now

File: android/hybridcommon-ui/src/main/java/com/revenuecat/purchases/hybridcommon/ui/PaywallHelpers.kt (hunk 3 above)
Comment (vegaro) [reply android/hybridcommon-ui/src/main/java/com/revenuecat/purchases/hybridcommon/ui/PaywallHelpers.kt:68]: But I think I am going to add the type just in case

Comment (vegaro) [reply android/hybridcommon-ui/src/main/java/com/revenuecat/purchases/hybridcommon/ui/PaywallHelpers.kt]: Yes that's what I'm doing in the opened PR, but this helper in phc will always get an offering identifier, it will never be an offering

Comment (NachoSoto) [reply android/hybridcommon-ui/src/main/java/com/revenuecat/purchases/hybridcommon/ui/PaywallHelpers.kt]: Yup yup 👍🏻

**Review by NachoSoto — APPROVED** [review]

**Comment by NachoSoto** [comment]
Merging this, I hope that's ok 😇 
I want to make a release with this + #675 before releasing #672.
